azurerm/resource_arm_recovery_services_protection_container.go

```diff
@@ -0,0 +1,148 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/recoveryservices/mgmt/2018-01-10/siterecovery"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmRecoveryServicesProtectionContainer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmRecoveryServicesProtectionContainerCreate,
+		Read:   resourceArmRecoveryServicesProtectionContainerRead,
+		Update: nil,
+		Delete: resourceArmSiteRecoveryProtectionContainerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"recovery_vault_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"recovery_fabric_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+		},
+	}
+}
+
+func resourceArmRecoveryServicesProtectionContainerCreate(d *schema.ResourceData, meta interface{}) error {
+	resGroup := d.Get("resource_group_name").(string)
+	vaultName := d.Get("recovery_vault_name").(string)
+	fabricName := d.Get("recovery_fabric_name").(string)
+	name := d.Get("name").(string)
+
+	client := meta.(*ArmClient).recoveryServices.ProtectionContainerClient(resGroup, vaultName)
+	ctx := meta.(*ArmClient).StopContext
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, fabricName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing recovery services protection container: %+v", err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_resource_group", *existing.ID)
+		}
+	}
+
+	parameters := siterecovery.CreateProtectionContainerInput{
+		Properties: &siterecovery.CreateProtectionContainerInputProperties{},
+	}
+
+	future, err := client.Create(ctx, fabricName, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating recovery services protection container1: %+v", err)
```

Comment:
As with all errors could we add a bit more detail:
```suggestion
		return fmt.Errorf("Error creating recovery services protection container %s (fabric %s): %+v", name, fabricName, err)
```